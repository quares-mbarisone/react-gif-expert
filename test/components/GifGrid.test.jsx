import { render, screen } from "@testing-library/react"
import { GifGrid } from "../../src/components/GifGrid"
import { useFetchGifs } from "../../src/hooks/useFetchGifs";
jest.mock('../../src/hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'One Piece';
    const gifs = [
        {
            id:'1',
            title: 'Luffy',
            url:'https://localhost/luffy.jpg'
        },
        {
            id:'2',
            title: 'Zoro',
            url:'https://localhost/zoro.jpg'
        },
        {
            id:'3',
            title: 'Nami',
            url:'https://localhost/nami.jpg'
        }
    ]

    
    test('debe de mostrar el loading inicialmente', () => { 
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        })
        render(<GifGrid category={ category }/>);
        expect( screen.getByText('Cargando...') );
        expect( screen.getByText(category) );
    });

    test('debe de mostrar items cuando se cargan las imÃ¡genes mediante el UseFetchGifs', () => {
        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        })
        render(<GifGrid category={ category }/>);
        expect( screen.getAllByRole('img').length ).toBe(3)
    })

    test('debe de hacer match con la snapshot', () => {
        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        })
        const { container } = render( <GifGrid category={ category }/> )
        expect( container ).toMatchSnapshot();
    })
})