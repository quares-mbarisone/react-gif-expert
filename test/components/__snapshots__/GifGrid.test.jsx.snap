// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de hacer match con la snapshot 1`] = `
<div>
  <h3>
     
    One Piece
     
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Luffy"
        src="https://localhost/luffy.jpg"
      />
      <p>
         
        Luffy
         
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Zoro"
        src="https://localhost/zoro.jpg"
      />
      <p>
         
        Zoro
         
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Nami"
        src="https://localhost/nami.jpg"
      />
      <p>
         
        Nami
         
      </p>
    </div>
  </div>
</div>
`;
